src/Compilers/CSharp/Test/Semantic/Semantics/NativeIntegerTests.cs

```diff
@@ -14180,5 +14180,189 @@ .maxstack  1
             verifier.VerifyIL("NativeInts.Checked4", expectedCheckedIL);
             verifier.VerifyIL("NativeInts.Checked5", expectedCheckedIL);
         }
+                             
+        [WorkItem(47887, "https://github.com/dotnet/roslyn/issues/47887")]
+        [Fact]
+        public void ArrayAccessNInt()
+        {
+            string source =
+@"class C
+{
+    static void Main()
+    {
+        int r = F(new int[] { 1 }, 0);
+        System.Console.WriteLine(r);
+    }
+    static int F(int[] a, nint i)
+    {
+        return a[i];
+    }
+}";
+            var verifier = CompileAndVerify(source, expectedOutput: "1", verify: Verification.Skipped);
+            verifier.VerifyIL("C.F",
+@"{
+  // Code size        4 (0x4)
+  .maxstack  2
+  IL_0000:  ldarg.0
+  IL_0001:  ldarg.1
+  IL_0002:  ldelem.i4
+  IL_0003:  ret
+}");
+        }
+
+        [WorkItem(47887, "https://github.com/dotnet/roslyn/issues/47887")]
+        [Fact]
+        public void ArrayAccessNUInt()
+        {
+            string source =
+@"class C
+{
+    static void Main()
+    {
+        int r = F(new int[] { 1 }, 0);
+        System.Console.WriteLine(r);
+    }
+    static int F(int[] a, nuint i)
+    {
+        return a[i];
+    }
+}";
+            var verifier = CompileAndVerify(source, expectedOutput: "1", verify: Verification.Skipped);
+            verifier.VerifyIL("C.F",
+@"{
+  // Code size        4 (0x4)
+  .maxstack  2
+  IL_0000:  ldarg.0
+  IL_0001:  ldarg.1
+  IL_0002:  ldelem.i4
+  IL_0003:  ret
+}");
+        }
+
+        [WorkItem(47887, "https://github.com/dotnet/roslyn/issues/47887")]
+        [Fact]
+        public void ArrayCreationNInt()
+        {
+            string source =
+@"class C
+{
+    static void Main()
+    {
+        const nint length = 3;
+        int[] a = new int[length];
```

Comment:
>length [](start = 26, length = 6)

Are we testing non-const scenario?

Reply:
Updated to test both const and non-const scenarios for all 6 tests.